sycl/include/sycl/ext/oneapi/atomic_enums.hpp

```diff
@@ -24,31 +24,40 @@ namespace sycl {
 namespace ext {
 namespace oneapi {
 
-using memory_order = cl::sycl::memory_order;
-__SYCL_INLINE_CONSTEXPR memory_order memory_order_relaxed =
-    memory_order::relaxed;
-__SYCL_INLINE_CONSTEXPR memory_order memory_order_acquire =
-    memory_order::acquire;
-__SYCL_INLINE_CONSTEXPR memory_order memory_order_release =
-    memory_order::release;
-__SYCL_INLINE_CONSTEXPR memory_order memory_order_acq_rel =
-    memory_order::acq_rel;
-__SYCL_INLINE_CONSTEXPR memory_order memory_order_seq_cst =
-    memory_order::seq_cst;
+using memory_order __SYCL2020_DEPRECATED("use 'sycl::memory_order' instead") =
+    cl::sycl::memory_order;
+__SYCL_INLINE_CONSTEXPR memory_order memory_order_relaxed __SYCL2020_DEPRECATED(
+    "use 'sycl::memory_order_relaxed' instead") = memory_order::relaxed;
+__SYCL_INLINE_CONSTEXPR memory_order memory_order_acquire __SYCL2020_DEPRECATED(
+    "use 'sycl::memory_order_acquire' instead") = memory_order::acquire;
+__SYCL_INLINE_CONSTEXPR memory_order memory_order_release __SYCL2020_DEPRECATED(
+    "use 'sycl::memory_order_release' instead") = memory_order::release;
+__SYCL_INLINE_CONSTEXPR memory_order memory_order_acq_rel __SYCL2020_DEPRECATED(
+    "use 'sycl::memory_order_acq_rel' instead") = memory_order::acq_rel;
+__SYCL_INLINE_CONSTEXPR memory_order memory_order_seq_cst __SYCL2020_DEPRECATED(
+    "use 'sycl::memory_order_seq_cst' instead") = memory_order::seq_cst;
 
-using memory_scope = cl::sycl::memory_scope;
-__SYCL_INLINE_CONSTEXPR memory_scope memory_scope_work_item =
-    memory_scope::work_item;
-__SYCL_INLINE_CONSTEXPR memory_scope memory_scope_sub_group =
-    memory_scope::sub_group;
-__SYCL_INLINE_CONSTEXPR memory_scope memory_scope_work_group =
-    memory_scope::work_group;
-__SYCL_INLINE_CONSTEXPR memory_scope memory_scope_device = memory_scope::device;
-__SYCL_INLINE_CONSTEXPR memory_scope memory_scope_system = memory_scope::system;
+using memory_scope __SYCL2020_DEPRECATED("use 'sycl::memory_scope' instead") =
+    cl::sycl::memory_scope;
+__SYCL_INLINE_CONSTEXPR
+memory_scope memory_scope_work_item __SYCL2020_DEPRECATED(
+    "use 'sycl::memory_scope_work_item' instead") = memory_scope::work_item;
+__SYCL_INLINE_CONSTEXPR
+memory_scope memory_scope_sub_group __SYCL2020_DEPRECATED(
+    "use 'sycl::memory_scope_sub_group' instead") = memory_scope::sub_group;
+__SYCL_INLINE_CONSTEXPR memory_scope memory_scope_work_group
+    __SYCL2020_DEPRECATED("use 'sycl::memory_scope_work_group' instead") =
+        memory_scope::work_group;
+__SYCL_INLINE_CONSTEXPR memory_scope memory_scope_device __SYCL2020_DEPRECATED(
+    "use 'sycl::memory_scope_device' instead") = memory_scope::device;
+__SYCL_INLINE_CONSTEXPR memory_scope memory_scope_system __SYCL2020_DEPRECATED(
+    "use 'sycl::memory_scope_system' instead") = memory_scope::system;
 
 #ifndef __SYCL_DEVICE_ONLY__
 namespace detail {
 
+__SYCL2020_DEPRECATED(
+    "use 'sycl::detail::getStdMemoryOrder(sycl::memory_order)' instead")
```

Comment:
Wow, that is something new to me. In which cases such diagnostic is going to be visible? I mean when user compiles their SYCL application, our headers are marked as system ones and they shouldn't produce "internal" diagnostics, i.e. warnings for things which are not directly used in user code.

Will it be visible during DPC++ RT build? If so, do we have a `-Werror` build which will fail (@alexbatashev?) ?

Reply:
As I see, the [[deprecated]] attribute for the alias introduced by the `using` keyword will be ignored by the compiler, since the compiler just substitutes the target of the alias. So this concrete deprecation is for documentation purposes only. But the [[deprecated]] attribute next to the `memory_order_relaxed` and the other constants generates a warning for the user, please have a look at the test https://github.com/psamolysov-intel/llvm/blob/make-sycl-atomic-ref-available/sycl/test/basic_tests/ext-atomic-ref-instantiation.cpp

Reply:
> Will it be visible during DPC++ RT build?

No, we disable all deprecation warnings during runtime build

Reply:
> No, we disable all deprecation warnings during runtime build

What is the point then of this deprecation? It won't help find usage of deprecated internal function? Or am I missing something?

Reply:
Probably it should be removed as user shouldn't use "private" API under sycl::detail:: namespace.

Reply:
@dm-vodopyanov Let's keep it there since just a marker the API is deprecated. In the future, the entire header will be removed as deprecated.